0000-constrained-backends.md

```diff
@@ -0,0 +1,110 @@
+Currently the publisher_acl is only applied via the salt api - but if a state enters the salt master via a (git) backend, there is no constraint checking, allowing anyone that can change an backend to basically change almost everything, if not everything.
+For example: This hinders to allow "foreign" git repositories to be used as backend without extensive merge approval process.
+
+In the end, an application administrator could contribute states / functions / ... that are specific to a software and enable more complex orchestrations, without having a merge approver to know everything about every application to spot (unintentional or intentional) malicious changes and an expert keeps being an expert for just one/assigned scope/s, without the need to be trusted with everything.
+
+# Design
+[design]: #detailed-design
+
+This is the bulk of the SEP. Explain the design in enough detail for somebody familiar
+with the product to understand, and for somebody familiar with the internals to implement. It should include:
+
+- Definition of any new terminology
+  - "foreign" backend: a backend that can be modified by other users than the salt master administrator.
+- Examples of how the feature is used:
+      - Target minions on backends
+        Users able to change webserver.git aren't always the same users that should be able to modify backupservers.
+      - Apply these states constrained to be applied as user x.
+        - This and more could be accomplished by forcing a render pipeline, that acts as gateway that applies or filters wrapping the renderers with another renderer, that applies/filters parts. Or simply raise an exception to stop further processing of a configuration state.
+      - Allow just a certain list of renderers.
+         - Some renderers  (like py) might have more capabilities than others to try to circumvent pillar scopes.
+
+
+      ```
+                gitfs_remotes:
+                  - https://foo.com/main.git
+                  - https://foo.com/webserver.git:
+                    - target: web-group-a\*
+                    - renderer-prepend: malware_scan
+                    - renderer-append: force_user_x
+                    - renderer-allow:
+                      -  jinja|yaml
+                  - https://foo.com/webserver.git:
+                    - target: web-group-b\*
+                    - renderer-append: force_user_y
+                    - renderer-allow:
+                      -  jinja|yaml
+                      -  py
+                  - https://foo.com/backups.git:
+                    - target: backup\*
+      ```
+  
+  The last part, is still unpolished:
+      * Render (some) state files in a sandbox.
+         * Rendering itself can be harmful on the salt master, as the renderer (for example: py) might get access to data / pillars that is not in scope for the targeted minion. Restricting the information that is available.
```

Comment:
This isn't true. State rendering happens on the minion. There's no way to access other minions' pillars.

Reply:
Sorry, my mixup. I corrected it.
Had originally in mind to call the minion a submaster of sorts.

Reply:
No, you've not corrected it. That's not how Salt works.

Reply:
https://github.com/monkz/salt-enhancement-proposals/commit/95946d9128cca982f0aff53128c38ef2cb5d7076

Reply:
Seems like Github is not updating this PR with the latest commits i've pushed to my fork's master.

Reply:
Found a solution for githubs lack of PR updates: https://github.com/orgs/community/discussions/16351#discussioncomment-6327770 Sorry for the notification noise.